src/core/model/table/dynamic_position_list_index.cpp

```diff
@@ -0,0 +1,200 @@
+#include "model/table/dynamic_position_list_index.h"
+
+#include <algorithm>
+#include <chrono>
+#include <cmath>
+#include <map>
+#include <memory>
+#include <stack>
+#include <utility>
+
+#include <boost/dynamic_bitset.hpp>
+#include <easylogging++.h>
+
+namespace {
+// O(delete_data * log(cluster))
+void ProcessDeleteToCluster(model::DynamicPositionListIndex::Cluster& cluster,
+                            std::vector<size_t> const& deleted_record_ids,
+                            std::vector<bool>& is_deleted) {
```

Comment:
Make `deleted_record_ids` an `unordered_set`, use `std::erase_if` with `deleted_record_ids.erase(record_id) == deleted_record_ids.end()` predicate. `std::erase_if` will maintain the sort invariant too.

Reply:
> Make `deleted_record_ids` an `unordered_set`

Why `unordered_set`? We do not process any lookup over `deleted_record_ids` (at least now, with `std::erase_if `realization)

Reply:
> Make `deleted_record_ids` an `unordered_set`

When i tried this, I had this error:
`ConfigurationError: Unable to cast Python object to C++ type for option "delete"`
As temporary solution, we construct `unordered_set` from `vector`, and pass it to `DynPLI::UpdateWith` Method

Reply:
That's because `std::unordered_set` is missing from `py_to_any.h`